### PR TITLE
make usage of web client more consistent including logging

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

**Comment by snopoke** [comment]
Build failure seems unrelated:
```
org.javarosa.xpath.test.XPathEvalTest > doTests FAILED
    org.javarosa.core.services.locale.LocaleTextException at XPathEvalTest.java:67
```

I often get this locally too. @shubham1g5 any idea why this happens?

File: src/main/java/org/commcare/formplayer/web/client/WebClient.java [hunk 1]
```diff
@@ -36,49 +40,63 @@ public <T> T get(URI uri, Class<T> responseType) {
...
-        );
+        ResponseEntity<String> entity = postRaw(uri, restoreFactory.getRequestHeaders(uri), body, String.class);
         Boolean shouldSync = true;
         if (entity != null && entity.getStatusCode() == HttpStatus.NO_CONTENT) {
             shouldSync = false;
         }
         return shouldSync;
     }
 
+    public <T, R> ResponseEntity<R> postRaw(URI uri, HttpHeaders headers, T body, Class<R> responseType) {
```
Comment (shubham1g5) [inline src/main/java/org/commcare/formplayer/web/client/WebClient.java:84]: 💯

**Review by shubham1g5 — APPROVED** [review]
looks great.

**Comment by shubham1g5** [comment]
It seems like that test needs to have localization configured to get the value for `ethiopian_months` . I am not sure how was it working so far (probably some static caching), I am looking at it and hopefully will PR a fix soon.

**Pull request merged**